### DISCUSSION TITLE
using `shouldTouch` and `shouldDirty` only sets touched

### DISCUSSION THREAD

**Discussion by HexSleeves** [react-hook-form/react-hook-form#7505] [RFC]
If I use `setValue` with the options `{ shouldTouch: true, shouldDirty: true }` it only sets the field as touched instead of dirty. Is this a bug?

https://codesandbox.io/s/react-hook-form-v7-ts-setvalue-forked-u8wsz?file=/src/index.tsx

**Comment by bluebill1049** [comment]
missing `defaultValues` please refer to the doc: https://react-hook-form.com/api/useform/formstate

**Reply by HexSleeves** [reply]
I reviewed the docs and updated the example to include defaultValues, yet it still does not work.

**Reply by Moshyfawn** [reply]
Yeah, it feels like a bug to me too. I tried re-creating it in this [codesandbox](https://codesandbox.io/s/should-dirty-fail-qnsfm); you can see in the devtools the `isDirty` field doesn't change when first or last name set with a click of an option button, tho the `souldDirty` flag is clearly set to `true`.
I quickly glanced over the source code. If [this](https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/createFormControl.ts#L558-L559) is the relevant piece of code to what the question here is, it looks like `shouldDirty` doesn't get used in `updateTouchAndDirty` function.

**Reply by bluebill1049** [reply]
can you raise an issue on this one?

**Reply by Moshyfawn** [reply]
#7506 issue created
